app/Models/Movie.php

```diff
@@ -34,6 +34,7 @@ public function addTrailer(string $string_url) {
     }
     public function addRating(User $user, int $rating) {
         if ($user instanceof User && is_int($rating)) {
+            $this->ratings()->detach($user->id);
```

Comment:
Why first detach. And then attach again?

Reply:
In the case there the user already have rated the movie